chromadb/test/configurations/test_collection_configuration.py

```diff
@@ -0,0 +1,224 @@
+import pytest
+from typing import Dict, Any, cast
+import numpy as np
+from chromadb.config import System
+from chromadb.api.types import (
+    EmbeddingFunction,
+    Embeddings,
+    Space,
+    Embeddable,
+)
+from chromadb.api.client import Client as ClientCreator
+from chromadb.api import ClientAPI
+from chromadb.api.collection_configuration import (
+    CreateCollectionConfiguration,
+    load_collection_configuration_from_json,
+    CreateHNSWConfiguration,
+)
+
+
+class LegacyEmbeddingFunction(EmbeddingFunction[Embeddable]):
+    def __init__(self) -> None:
+        pass
+
+    def __call__(self, input: Embeddable) -> Embeddings:
+        return cast(Embeddings, np.array([[1.0, 2.0]], dtype=np.float32))
+
+    @staticmethod
+    def name() -> str:
+        return "legacy_ef"
+
+
+class CustomEmbeddingFunction(EmbeddingFunction[Embeddable]):
+    def __init__(self, dim: int = 3):
+        self._dim = dim
+
+    def __call__(self, input: Embeddable) -> Embeddings:
+        return cast(Embeddings, np.array([[1.0] * self._dim], dtype=np.float32))
+
+    @staticmethod
+    def name() -> str:
+        return "custom_ef"
+
+    def get_config(self) -> Dict[str, Any]:
+        return {"dim": self._dim}
+
+    @staticmethod
+    def build_from_config(config: Dict[str, Any]) -> "CustomEmbeddingFunction":
+        return CustomEmbeddingFunction(dim=config["dim"])
+
+    def default_space(self) -> Space:
+        return Space.COSINE
+
+
+def test_legacy_embedding_function(client: ClientAPI) -> None:
+    """Test creating and getting collections with legacy embedding functions"""
+    client.reset()
+
+    # Create with legacy embedding function
+    coll = client.create_collection(
+        name="test_legacy",
+        embedding_function=LegacyEmbeddingFunction(),
+    )
+
+    # Verify the configuration marks it as legacy
+    config = load_collection_configuration_from_json(coll._model.configuration_json)
+    if config and isinstance(config, dict):
+        ef_config = config.get("embedding_function", {})  # type: ignore
+        if isinstance(ef_config, dict):
+            assert ef_config.get("type") == "legacy"
```

Comment:
it's possible for this assertion to never be evaluated if the if statements above evaluate to false
seems like if the above if statements evaluate to false this test should fail rather than passing?